### PR TITLE
New comment component

### PR DESCRIPTION
**Pull request by rtichauerv**
I created a dummy resource page that catches the resourceId, then a NewResourceComment component to call from this page, and a NewResourceCommentForm that this previous component uses to ask for the content of the comment and then submit it through the API.

### CODE REVIEW

File: components/NewResourceComment.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { OpenAPIProvider } from "react-openapi-client";
+import React from "react";
+import { Button, Modal, Box } from "@mui/material";
+import NewResourceCommentForm from "./NewResourceCommentForm";
+
+function NewResourceComment({ resourceId }) {
+  const [open, setOpen] = React.useState(false);
```
Comment (ristofer) [inline components/NewResourceComment.js:7]: It is better to import implicitly the useState() function. https://reactjs.org/docs/hooks-state.html

File: package.json [hunk 2]
```diff
@@ -14,16 +14,17 @@
     "@fontsource/roboto": "^4.5.8",
     "@mui/icons-material": "^5.10.6",
     "@mui/material": "^5.10.6",
+    "axios": "^0.27.2",
```
Comment (ristofer) [inline package.json:17]: This line must be deleted to prevent conflicts with the line 27

**Review by ristofer — CHANGES_REQUESTED** [review]

File: components/NewResourceCommentForm.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    return (
+      <Box sx={{ display: "flex" }}>
+        <CircularProgress />
+      </Box>
+    );
+  }
+  if (data) {
+    return (
+      <>
+        <h6>Comment created: {data.content}</h6>
```
Comment (iaheck) [inline components/NewResourceCommentForm.js:44]: this does not work in dark mode
image

File: components/NewResourceComment.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    transform: "translate(-50%, -50%)",
+    width: 600,
+    bgcolor: "background.paper",
+    border: "2px solid #000",
+    boxShadow: 24,
+    p: 4,
+  };
+
+  return (
+    <OpenAPIProvider definition="http://localhost:3001/api-docs/v1/swagger.json">
```
Comment (iaheck) [inline components/NewResourceComment.js:23]: we should use relative paths to avoid changing things when deploying. Just start with /api-docs(...) I think

**Review by iaheck — CHANGES_REQUESTED** [review]
It works! just some minor issues

File: components/NewResourceComment.js (hunk 1 above)
Comment (rtichauerv) [reply components/NewResourceComment.js]: Done!

File: package.json (hunk 2 above)
Comment (rtichauerv) [reply package.json]: Done!

File: components/NewResourceCommentForm.js (hunk 3 above)
Comment (rtichauerv) [reply components/NewResourceCommentForm.js]: Fixed!

File: components/NewResourceComment.js (hunk 4 above)
Comment (rtichauerv) [reply components/NewResourceComment.js]: Fixed as you said, with the relative path only

**Review by ristofer — APPROVED** [review]

**Review by iaheck — APPROVED** [review]

**Pull request merged**